test/query/shorthand.test.ts

```diff
@@ -33,6 +35,159 @@ describe('query/shorthand', () => {
+        shorthandParser.fn(encQ, 'count(*,q)');
+        assert.deepEqual(encQ,{aggregate: AggregateOp.COUNT, field: '*', type: Type.QUANTITATIVE});
+      });
+
+      it('should correctly parse an encoding query given a fieldDefShorthand with timeunit function', () => {
+        let encQ: EncodingQuery = {} as EncodingQuery;
+        shorthandParser.fn(encQ, 'hours(a,t)');
+        assert.deepEqual(encQ, {field: 'a', timeUnit: TimeUnit.HOURS, type: Type.TEMPORAL});
+      });
+
+      it('should correctly parse an encoding query given a fieldDefShorthand with bin function', () => {
+        let encQ: EncodingQuery = {} as EncodingQuery;
+        shorthandParser.fn(encQ, 'bin(a,q,maxbins=20)');
+        assert.deepEqual(encQ, {
+          bin: {maxbins: 20},
+          field: 'a',
+          type: Type.QUANTITATIVE,
+        });
+      });
+    });
+
+    describe('rawFieldDef', () => {
+      it('should correctly parse an encoding query from fieldDef parts', () => {
+        let encQ = shorthandParser.rawFieldDef({} as EncodingQuery,
+          splitWithTail('a,q,scale={"domain":[1,2],"exponent":3,"type":"pow"},axis={"orient":"top"}', ',', 2)
+        );
+        assert.deepEqual(encQ, {axis: {orient: AxisOrient.TOP}, field: 'a', scale: {domain: [1, 2], exponent: 3, type: ScaleType.POW}, type: Type.QUANTITATIVE});
+      });
+
+      it('should correctly parse an encoding query from fieldDef parts', () => {
+        let encQ = shorthandParser.rawFieldDef({} as EncodingQuery,
+          splitWithTail('a,n,sort={"field":"a","op":"mean","order":"descending"}', ',', 2)
+        );
+        assert.deepEqual(encQ, {field: 'a', sort: {field: 'a', op: AggregateOp.MEAN, order: SortOrder.DESCENDING}, type: Type.NOMINAL});
+      });
+    });
+
+    describe('calculate', () => {
+      it('should correctly parse calculate from shorthand', () => {
+        let specQ: SpecQuery = {} as SpecQuery;
+        // FIXME(https://github.com/uwdata/compassql/issues/260)
+        let shorthandPart = 'calculate:{x2:datum.x*2},{x3:datum.x*3}';
+        let splitPartValue = splitWithTail(shorthandPart, ':', 1)[1];
+
+        shorthandParser.calculate(specQ, splitPartValue);
+        assert.deepEqual(specQ.transform.calculate, [{field: 'x2', expr: 'datum.x*2'}, {field: 'x3', expr: 'datum.x*3'}]);
+      });
+    });
+
+    describe('filter', () => {
+      it('should correctly parse filter from shorthand', () => {
+        let specQ: SpecQuery = {} as SpecQuery;
+        let shorthandPart = 'filter:[{"field":"color","equal":"red"},"datum[\\"b2\\"] > 60",{"field":"color","oneOf":["red","yellow"]},{"field":"x","range":[0,5]}]';
+        let splitPartValue = splitWithTail(shorthandPart, ':', 1)[1];
+
+        shorthandParser.filter(specQ, splitPartValue);
+        assert.deepEqual(specQ.transform.filter, [
+          {field: 'color', equal: 'red'}, 'datum["b2"] > 60', {field: 'color', oneOf: ['red', 'yellow']}, {field: 'x', range: [0,5]}
```

Comment:
You can't really specify filter as `datum["b2"] > 60` in JSON?

It has to be one of
- `datum.b2 > 60`
- `datum['b2'] > 60`
- `datum[\"b2\"] > 60`